### PR TITLE
SFx exporter: Implement summary type

### PR DESCRIPTION
**Pull request by pmcollins**
Prometheus's Summary type was previously not implemented in the SFx exporter.
This change ports the SFx Smart Agent's implementation to the SFx exporter. It has
been end-to-end tested using a Prometheus server as a source of metrics
sent to an SFx backend.

Addresses https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/2944

### CODE REVIEW

File: exporter/signalfxexporter/translation/converter.go [hunk 1]
```diff
@@ -152,6 +154,60 @@ func labelsToDimensions(labels pdata.StringMap, extraDims []*sfxpb.Dimension) []
...
+		for j := 0; j < qvs.Len(); j++ {
+			qPt := sfxpb.DataPoint{
+				Metric:     name + "_quantile",
+				Timestamp:  ts,
+				MetricType: &sfxMetricTypeGauge,
+			}
+			qv := qvs.At(j)
+			qdim := sfxpb.Dimension{
+				Key:   "quantile",
+				Value: strconv.FormatFloat(qv.Quantile(), 'f', 6, 64),
```
Comment (rmfitzpatrick) [inline exporter/signalfxexporter/translation/converter.go:200]: Six digits of precision seems a little high to me for quantile in most cases ("0.250000" when "0.25" is likely desired).  Does TrimRight(quantile, "0") make sense?

**Review by jrcamp — APPROVED** [review]

File: exporter/signalfxexporter/translation/converter.go (hunk 1 above)
Comment (pmcollins) [reply exporter/signalfxexporter/translation/converter.go:200]: Hmm, I'm not sure. This line was copied from SA -- figured we'd want to keep it the same
https://github.com/signalfx/signalfx-agent/blob/c295133b4a52b2fff071c8070536183a48e403fc/pkg/monitors/prometheusexporter/conversion.go#L77

Comment (rmfitzpatrick) [reply exporter/signalfxexporter/translation/converter.go:200]: I'm not sure there would be any compatibility guarantees from this monitor to the exporter, but can see it either way.

**Review by rmfitzpatrick — APPROVED** [review]

File: exporter/signalfxexporter/translation/converter.go (hunk 1 above)
Comment (pmcollins) [reply exporter/signalfxexporter/translation/converter.go:200]: Apologies, I didn't mean to dismiss your idea -- I'm partial to it. @keitwb do you know where the six digits of precision came from?

Comment (keitwb) [reply exporter/signalfxexporter/translation/converter.go:200]: Seems like `-1` would be the best option for the `prec` argument.  That makes it use the least number of decimal points required.  I can't remember why -6 was used, might have been no reason at all.

Comment (pmcollins) [reply exporter/signalfxexporter/translation/converter.go]: Thank you, done.

**Pull request merged**